src/devicewrapper.cpp

```diff
@@ -0,0 +1,66 @@
+/* ****************************************************************************
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+*
+*
+* This file is part of Microsoft R Host.
+*
+* Microsoft R Host is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Microsoft R Host is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Microsoft R Host.  If not, see <http://www.gnu.org/licenses/>.
+*
+* ***************************************************************************/
+
+#include "devicewrapper.h"
+#include "msvcrt.h"
+
+namespace rhost {
+    namespace grdevices {
+        devicewrapper::devicewrapper() {
+            _ver = R_GE_getVersion();
+        }
+
+
+        devicewrapper::devicewrapper(pDevDesc dd) {
+            _ver = R_GE_getVersion();
+            switch (_ver) {
```

Comment:
>switch [](start = 12, length = 6)

Assert in default case and always initialize both pointers (set the unused one to nullptr). If we needed to save memory, we could use a union since only one of dd_11 and dd_12 will ever be used, but union is more bug prone and this is not a memory intensive scenario.